### PR TITLE
chore: simplify BumpUps deduplication

### PR DESCRIPTION
**Pull request by romange**
This pr #2474 introduced iterator protection by
tracking which keys where bumped up during the transaction operation. This was done by managing keys view set. However, this can be simplified using fingerprints. Also, fingerprints do not require that the original keys exist.

In addition, this #3241 PR introduces FetchedItemsRestorer that tracks bumped set and saves it to protect against fiber context switch. My claim is that it's redundant. Since we only keep the auto-laundering iterators, when a fiber preempts these iterators recognize it (see IteratorT::LaunderIfNeeded) and refresh themselves anyway.

To summarize: fetched_items_ protect us from iterator invalidation during atomic scenarios, and auto-laundering protects us from everything else, so fetched_items_ can be cleared in that case.

### CODE REVIEW

File: src/server/db_slice.cc [hunk 1]
```diff
@@ -221,18 +207,18 @@ unsigned PrimeEvictionPolicy::Evict(const PrimeTable::HotspotBuckets& eb, PrimeT
 // because some other transaction might conclude and clear the fetched_items_ with OnCbFinish()
 class FetchedItemsRestorer {
  public:
-  using RestoreType = absl::flat_hash_set<CompactObjectView>;
-  explicit FetchedItemsRestorer(RestoreType* dst) : dst_to_restore_(dst) {
-    cached_ = std::move(*dst_to_restore_);
+  // using RestoreType = absl::flat_hash_set<CompactObjectView>;
```
Comment (romange) [inline src/server/db_slice.cc:210]: this code should be deleted. hacking it as a noop to get the initial feedback for review. @adiholden  @chakaz  FYI

Comment (adiholden) [reply src/server/db_slice.cc:210]: As far as I understand this code comes to protect from DbSlice::OnCbFinish calling fetched_items_.clear to not clear the fetched items in caller transaction while the preempt
OnCbFinish is called on each transaction finish so if we have 2 interliving transactions one running the OnCbFinish after a single key was fetched before another one was fetched it will clear the vector not as expected

Comment (romange) [reply src/server/db_slice.cc:210]: Have you read the PR description? My claim is that if a tx preempts we do not need to protect iterators - and that was the purpose of `fetched_items_`

Comment (adiholden) [reply src/server/db_slice.cc:210]: so why did you keep fetched_items_ logic and moved to finger print ? if this is the claim we can remove fetched_items_ at all

File: src/server/db_slice.h [hunk 2]
```diff
@@ -592,14 +592,20 @@ class DbSlice {
...
   // Used in temporary computations in Acquire/Release.
-  mutable absl::flat_hash_set<uint64_t> uniq_fps_;
+  mutable absl::flat_hash_set<uint64_t, FpHasher> uniq_fps_;
 
   // ordered from the smallest to largest version.
   std::list<std::pair<uint64_t, ChangeCallback>> change_cb_;
 
   // Used in temporary computations in Find item and CbFinish
-  mutable absl::flat_hash_set<CompactObjectView> fetched_items_;
+  mutable absl::flat_hash_set<uint64_t, FpHasher> fetched_items_;
```
Comment (adiholden) [inline src/server/db_slice.h:608]: lets add a comment why we need this hashset , why we can not bumpup item twice in transaction without preemption but ok to to bumpup twice when we do preempt

Comment (adiholden) [reply src/server/db_slice.h:608]: expect this comment the changes looks ok and we can merge

**Review by adiholden — APPROVED** [review]
